lib/bundler/resolver.rb

```diff
@@ -324,8 +324,8 @@ def verify_gemfile_dependencies_are_found!(requirements)
             end
           else
             message = "Could not find gem '#{requirement}' "
-            if @index.source_types.include?(Bundler::Source::Rubygems)
-              message << "in any of the gem sources listed in your Gemfile."
+            if Bundler.definition.has_rubygems_remotes?
```

Comment:
I'd prefer we didn't reference the `Bundler.definition` when not necessary, as it makes this method reliant on global state